### ISSUE TITLE
NaN loss early in training

### ISSUE THREAD

**Issue by tinazezlin** [Y-debug-sys/Diffusion-TS#112]
I followed the official tutorial using sine data, but encountered a problem during training: the loss becomes NaN very early. All generated synthetic data is also NaN.

Upon investigation, the NaNs in the loss appear to be caused by NaN gradients in several model parameters. PyTorch’s anomaly detection reported GeluBackward0 as the source of the first NaN.

Could you suggest what might be causing this problem and how to fix it?

**Comment by Y-debug-sys** [comment]
Are you training on CPU? Similar issues have occurred before (see https://github.com/Y-debug-sys/Diffusion-TS/issues/97 )—or you might want to try replacing GELU with ReLU to help identify the source of the problem.
